CONTRIBUTING.md

```diff
@@ -14,23 +14,31 @@ If you are planning to start some significant coding, you would benefit from ask
 
 1. [Fork](https://help.github.com/articles/fork-a-repo/) the repository and [clone](https://help.github.com/articles/cloning-a-repository/) your fork.
 
-2. Start coding!
+2. Before you start coding, open an [issue at Github](https://github.com/minetest/minetest/issues) to discuss your intended contribution with the core developers, this may save time for you and others, and may avoid wasted work.
+
```

Comment:
I'd advise making the issue sound more like a suggestion. ("open an issue..." is directly telling people to do it, even if the next bit implies it's optional.). Maybe "Consider opening an issue ... to discuss ..."? You should also look at the earlier part you haven't edited, which mentions discussing on IRC. This should all be combined into one part, with an issue or discussion via IRC suggested as alternative approaches. You might also add that it's worth searching existing issues?

New sentence should begin from "this may..."